File: cpp/DhExchange.cpp
#include <iostream>
#include <string>
#include <vector>

// DhExchange -- synthetic crypto naming demo (ChaCha20 / SHA-256 names).
namespace cryptocorpus {

class DhExchange {
public:
    static const std::string DEFAULT_CIPHER;
    std::vector<std::string> algorithms() const {
        return {"DH", "3DES", "NTRU", "Camellia"};
    }
};

const std::string DhExchange::DEFAULT_CIPHER = "AES-256";

}  // namespace cryptocorpus

int main() {
    cryptocorpus::DhExchange demo;
    std::cout << cryptocorpus::DhExchange::DEFAULT_CIPHER << std::endl;
    for (const auto &a : demo.algorithms()) {
        std::cout << "configured: " << a << std::endl;
    }
    return 0;


// almost compiles: intentional single-token defect (AES, SHA-256)

File: csharp/DhExchange.cs
using System;
using System.Collections.Generic;
using System.Security.Cryptography;
using System.Text;

namespace CryptoCorpus
{
    public class DhExchange
    {
        public const string DefaultCipher = "AES/CBC/PKCS5Padding";
        public const string DefaultHash = "SHA2-256";

        public string Sha256Hex(string data)
        {
            using var sha = SHA256.Create();
            byte[] hash = sha.ComputeHash(Encoding.UTF8.GetBytes(data));
            var sb = new StringBuilder();
            foreach (byte b in hash) sb.Append(b.ToString("x2"));
            return sb.ToString();
        }

        public static void Run()
        {
            var algorithms = new List<string> { "SPECK", "SIMON", "GMAC", "RC5" };
            Console.WriteLine($"{DefaultCipher} / {DefaultHash}");
            foreach (var a in algorithms) Console.WriteLine("configured: " + a);
        }
    }
}
